syntax = "proto2";

package DB.Protos;
option java_package = "org.byconity.proto";
option java_outer_classname = "HudiMeta";

message Column {
  optional string name = 1;
  optional string type = 2;
}

message Properties {
  message KeyValue {
    optional string key = 1;
    optional string value = 2;
  }

  repeated KeyValue properties = 1;
}

message HudiMetaClientParams
{
  optional Properties properties = 1;
}

message HudiTable {
  optional string hive_db_name = 1;
  optional string hive_table_name = 2;
  repeated Column columns = 3;
  repeated string partition_column_name = 4;
  optional Properties properties = 5;
}

message PartitionPaths {
  repeated string paths = 1;
}

message HudiFileSlices {
  message FileSlice {
    optional string base_file_name = 1;
    repeated string delta_logs = 2;
    optional uint64 base_file_length = 3;
  }
  repeated FileSlice file_slices = 1;
  optional string instant = 2;
}